tests/linen/module_test.py

```diff
@@ -668,6 +670,23 @@ def bar(self):
     with self.assertRaisesRegex(AttributeError, "has no attribute 'setup_called'"):
       empty.bar()
 
+  def test_module_with_attrs(self):
+    class Foo(nn.Module):
+      bar: nn.Dense = dataclasses.field(init=False)
+
+      def setup(self):
+        self.bar = nn.Dense(2)
+      
+      @nn.compact
```

Comment:
probably better to remove nn.compact here as it's unrelated to the thing being tested

Reply:
done